pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/ProtoWriterTools.java

```diff
@@ -259,29 +278,53 @@ private static void writeStringNoChecks(WritableSequentialData out, FieldDefinit
     }
 
     /**
-     * Write a bytes to data output
+     * Write a bytes to data output, assuming the corresponding field is non-repeated, and field type
+     * is any delimited: bytes, string, or message.
      *
      * @param out The data output to write to
-     * @param field the descriptor for the field we are writing
+     * @param field the descriptor for the field we are writing, the field must not be repeated
      * @param value the bytes value to write
      * @throws IOException If a I/O error occurs
      */
-    public static void writeBytes(WritableSequentialData out, FieldDefinition field, RandomAccessData value) throws IOException {
-        assert field.type() == FieldType.BYTES : "Not a byte[] type " + field;
+    public static void writeBytes(final WritableSequentialData out, final FieldDefinition field,
+            final RandomAccessData value) throws IOException {
+        assert field.type() == FieldType.BYTES ||
+                field.type() == FieldType.STRING ||
+                field.type() == FieldType.MESSAGE
+                : "Not a byte[] type " + field;
```

Comment:
I found one of `ProtoWriterTools` test cases calls this method (indirectly) for a string field, providing string bytes as a wrapped Bytes object. If assertions were enabled, this test case would fail.

Rather than removing or rewriting the test case, I decided to loose the assertion here. Why not allowing this method to be called for any delimited (bytes, string, message) field?

Reply:
An arbitrary bytes arrays may not necessarily represent a valid UTF-8 string. This method can successfully write the bytes array, but the deserialization operation may fail for the written bytes then. Same deal with messages.

I'd suggest to keep the assertion as it is and try and modify the test (and the code the test was written for) to avoid this risky code-path. If the code-path is still useful, then a new method should be introduced (similar to "NoChecks") that would bypass the validation. This would make the code that uses this code-path aware of what it's doing rather then implicitly assuming that everything should work, which it may not as explained above.

Reply:
OK, I reverted the assertion and fixed the test instead